src/strings/kmp_substring_search.jl

```diff
@@ -0,0 +1,100 @@
+"""
+    GetIndexWithKMP(string::String, sub_string::String, ignore_case::Bool)::Int
+
+Find if a string contain a given sub string with KMP Search, as an explanation might be too lengthy
+As such, a detailed explanation can be found at https://towardsdatascience.com/pattern-search-with-the-knuth-morris-pratt-kmp-algorithm-8562407dba5b
+
+
+Question
```

Comment:
I think it should be like this here, as you are providing an answer.
```suggestion
Questions and answers

```